docs/design/datacontracts/Thread.md

```diff
@@ -0,0 +1,195 @@
+# Contract Thread
+
+This contract is for reading and iterating the threads of the process.
+
+## Data structures defined by contract
+``` csharp
+record struct DacThreadStoreData (
+    int ThreadCount,
+    TargetPointer FirstThread,
+    TargetPointer FinalizerThread,
+    TargetPointer GcThread);
+
+record struct DacThreadStoreCounts (
+    int UnstartedThreadCount,
+    int BackgroundThreadCount,
+    int PendingThreadCount,
+    int DeadThreadCount);
+
+enum ThreadState
```

Comment:
@noahfalk, handling of enums and varying values between implementations is a problem. At the moment, since this enum is defined in the API portion of the contract, it does not represent the enum values as are present in the runtime at all. Instead it represents the enum values that the contract exposes. I really should have named this `DacThreadState` What we don't have in our specs is a manner to a contract from the runtime that indicates that the values of the runtimes variant of the `ThreadState` enum have some specific set of values.

Also, as @jkotas noted earlier, we probably shouldn't be exposing this complete set of enum values from the contracts. Instead we should have a restricted set that we believe is likely to be kept for the longer term exposed from the contracts, and prevent the contract from allowing visibility into other values held in the corresponding field in the runtime data structure.

We have the luxury of the contract api not being a public, stable api, so if we guess wrong on the set of exposed flags, we can adjust that, but it will still be somewhat of a pain to do that.